Src/App/Router.php

```diff
@@ -1,31 +1,37 @@
 <?php
 namespace Aot\App;
 
-use Aot\Controller\PostController;
 
 class Router
 {
-    private $pc_instance;
-
-
-    public function __construct()
-    {
-        $this->pc_instance = new PostController();
-    }
 
     /**
-     * select the method
-     * @param  array $datas
-     * @return array
+     * lead to the controller
+     * @param $parameters
+     * @return mixed
+     * @throws \Exception
      */
-    public function method($datas = [])
+    public function route($parameters)
     {
-        if ($datas['method'] == 'show' && !array_key_exists('id', $datas)){
-            return $this->pc_instance->index($datas);
-        } elseif ($datas['method'] == 'show') {
-            return $this->pc_instance->show($datas);
+
+        if (isset($parameters['controller']) && isset($parameters['method'])) {
+            $controlerName = '\Aot\Controller\\' . ucfirst($parameters['controller']) . 'Controller';
+            $method = $parameters['method'];
         } else {
-            throw new \Exception("not found");
+            throw new \Exception("method or controller invalid");
+        }
+        if (class_exists($controlerName)) {
+            $controller = new $controlerName();
+        } else {
+            throw new \Exception("controller invalid");
+        }
+        if (method_exists($controller, $method)) {
```

Comment:
If the `$method` is `protected`, the method exists but it is not callable. Please use `is_callable` instead